dockerfiles/ink-waterfall-ci/Dockerfile

```diff
@@ -52,13 +52,27 @@ RUN	set -eux; \
 # `--locked` ensures the project's `Cargo.lock` is used.
 	cargo install --git https://github.com/paritytech/canvas-node.git \
 		--locked --branch master && \
+# We additionally install the `canvas-node` as `canvas-rand-extension`.
+# This installation though is a modified `canvas-node`, so that ink!'s
+# `rand-extension` chain extension example is included in the runtime.
+# This enables us to test in the waterfall that the `rand-extension`
+# integration with Substrate still works.
+	git clone --depth 1 https://github.com/paritytech/canvas-node.git && \
+	curl -s https://raw.githubusercontent.com/paritytech/ink/master/examples/rand-extension/runtime/chain-extension-example.rs \
+		>> canvas-node/runtime/src/lib.rs && \
+	sed -i 's/type ChainExtension = ();/type ChainExtension = FetchRandomExtension;/g' canvas-node/runtime/src/lib.rs && \
+	sed -i 's/name = "canvas"/name = "canvas-rand-extension"/g' canvas-node/node/Cargo.toml && \
```

Comment:
I see the potential non-documented issues with this custom installation, maybe it makes sense to create a feature flag instead? That way it would be better supported.

Reply:
You mean a feature flag in the `canvas-node` code base, right? I had thought about this as well, but the thing is that everything regarding `ink-waterfall` is just meant for testing and I would rather not have anything regarding ink! examples in our production code base for `canvas-node`.

Reply:
And this kind of testing is only meant to be internal, right?

Reply:
Yup!